definitions/scenarios/satellite_upgrade.rb

```diff
@@ -72,6 +72,7 @@ class PreMigrations < Abstract
 
     def compose
       add_steps(
+        Procedures::Repositories::CleanupDuplicateErratumPackages,
```

Comment:
This is in Katello master (not even in 4.18 rc2), so I think it can't run as a pre-migration before the upgrade?

Reply:
We have the PR in the cherrypicks for 4.18 GA.. https://github.com/Katello/katello/pull/11494

Reply:
Ok, but then it still needs to be post-migrations to have 4.18 :)

Reply:
Oh..Are the pre-migrations run before the yum update? We want this step to run before the db:migrate on upgrades where the dup data causes errors..

Reply:
Yeah. You'd need to edit Migrations to do what you want, but it sounds dangerous to me, as at that moment the services will already be down and you'd have to restart them with the new code but old db.